gmp-droid.cpp

```diff
@@ -266,6 +266,8 @@ class DroidVideoDecoder : public GMPVideoDecoder
     if (m_codec) {
       ResetCodec ();
     }
+    m_draining = false;
+    m_resetting = false;
```

Comment:
Can the `m_resetting = true;` on line 263 be removed, or is there some multithreading magic going on here?

Reply:
That's the only place it's set to true I think. I need that boolean as ResetCodec can take a while to finish and I have to make sure that doesn't re-enter.

Reply:
Okay, that makes sense.

Reply:
Perhaps you should use something like `std::atomic<bool>` for `m_draining` to avoid optimisation+threading issues?